### DISCUSSION TITLE
Add developer support widget

### DISCUSSION THREAD

**Discussion by nmohoni** [npm/feedback#228] [Website]
Over 2,600 developers & contributors provide support via xs:code chat by adding a link on their ReadMe ([example](https://www.npmjs.com/package/vue-tables-2)).
I think that would be great to add the xs:code widget to the website, so users can get support for any package without leaving the nmp website.
It will also help the maintainers get the resources they need, and the motivation, to keep maintaining the packages and solve issues. 
We can solve the OS sustainability problem together 🥇 
Here's a demo:

![newpage2](https://user-images.githubusercontent.com/49695109/108173634-0eaf2080-7107-11eb-985c-743e7f0bbc8b.gif)

**Comment by nmohoni** [comment]
[Documentation link.
](https://xscodehelp.zendesk.com/hc/en-us/articles/360016587998-Chat-SDK-API-documentation)
[How it works](https://xscode.com/api-integration/)

**Comment by MylesBorins** [comment]
Hey @nmohoni. If developers can already add the link to their README what would you see as the significant advantage to built in support widget?

One concern I would have is the folks may think the support team they are talking to are GitHub or npm employees, which is not going to be the case.

In general I am hesitant about adding deep integration into the package landing page for 3rd party services, even more concerned if it involves 1:1 chat.

**Reply by nmohoni** [reply]
Hey @MylesBorins ,
Thanks for replying.
I understand the concern you're talking about (users might think it's Github/ npm employees), but I can also see think about several ways to ensure that they'll understand immediately that the support is offered by the maintainers. I think that's easy to solve in the UI.
I also understand why you hesitate to integrate a 3rd party in the landing pages.
I do think that this feature shouldn't be a built-in for all the packages, and what I'm offering is to open this ability for the maintainers/packages owners, and let them decide to add a chat button and support their users.
It appears from our research (we've been interviewed over 300 OSS developers) that over 60% of them would add a "paid support" feature if that was a built-in solution.
I honestly believe that it can help both the maintainers and the developers who use their code and make open-source sustainable.
Perhaps the right way is to add this ability for a limited amount of users and run some A/B testing?

**Reply by MylesBorins** [reply]
@nmohoni can you expand on why this needs to be built in to npm rather than a button included in a package author's readme?

**Reply by nmohoni** [reply]
@MylesBorins Sure.
1. That's one of the main pieces of feedback we got on our survey from both maintainers and developers who use npm: Developers who need the support are intuitively not going back to the README file. they're googling the errors/searching for solutions on Q&A sites. Over 82% of the developers we interviewed said that it will make much more sense to have this ability on npm / on their IDE (that's one of the reasons why we've developed also the [Aces VScode Extention](https://xscode.com/vscode-extension/))
2. Over 62% of the maintainers said that they preferred to leave their README file "clean".
3. Trust: 74% of the developers and R&D leaders said that if they'll be able to see on npm if a specific package has a support tier, it will make their decision easier when they're debating which package to use. When the data is on the README - it's harder to compare.
